File: java/jimmer-core/src/main/java/org/babyfish/jimmer/example/core/model/Book.java
package org.babyfish.jimmer.example.core.model;

import org.babyfish.jimmer.Immutable;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.List;

@Immutable
public interface Book {

    String name();

    @Nullable
    BookStore store();

    int price();

    LocalDateTime lastModifiedTime();

    List<Author> authors();
}

File: kotlin/jimmer-core-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/core/model/Book.kt
package org.babyfish.jimmer.example.kt.core.model

import org.babyfish.jimmer.Immutable
import java.time.LocalDateTime

@Immutable
interface Book {

    val name: String

    val store: BookStore?

    val price: Int

    val lastModifiedTime: LocalDateTime

    val authors: List<Author>
}